src/main/scala/com/databricks/labs/mosaic/expressions/geometry/ST_Centroid.scala

```diff
@@ -53,25 +55,33 @@ case class ST_Centroid(inputGeom: Expression, geometryAPIName: String, nDim: Int
           ev,
           eval => {
               val geometryAPI = GeometryAPI.apply(geometryAPIName)
-              val (inCode, geomInRef) = ConvertToCodeGen.readGeometryCode(ctx, eval, inputGeom.dataType, geometryAPI)
-              val (centroidCode, centroidRow) = CentroidCodeGen(geometryAPI).centroid(ctx, geomInRef, geometryAPI, nDim)
-
-              geometryAPIName match {
-                  case "ESRI" => s"""
-                                   |$inCode
-                                   |$centroidCode
-                                   |${ev.value} = $centroidRow;
-                                   |""".stripMargin
-                  case "JTS" => s"""
-                                   |try {
-                                   |$inCode
-                                   |$centroidCode
-                                   |${ev.value} = $centroidRow;
-                                   |} catch (Exception e) {
-                                   | throw e;
-                                   |}
-                                   |""".stripMargin
-
+              val tryIO = Try {
+                  val (inCode, geomInRef) = ConvertToCodeGen.readGeometryCode(ctx, eval, inputGeom.dataType, geometryAPI)
+                  val (centroidCode, centroidRow) = CentroidCodeGen(geometryAPI).centroid(ctx, geomInRef, geometryAPI, nDim)
+                  ((inCode, geomInRef), (centroidCode, centroidRow))
+              }
+              (tryIO, geometryAPI) match {
+                  case (
+                        Success(((inCode, _), (centroidCode, centroidRow))),
+                        ESRI
+                      ) => s"""
+                              |$inCode
+                              |$centroidCode
+                              |${ev.value} = $centroidRow;
+                              |""".stripMargin
+                  case (
+                        Success(((inCode, _), (centroidCode, centroidRow))),
+                        JTS
+                      ) => s"""
+                              |try {
```

Comment:
because generated java code wont compile without throws statement. 
JTS actually can throw an exception in any of these expressions and the signatures of these functions have throws declaration while ESRI can throw the exceptions but it doesnt have the throws declartion.
We do not wrap ESRI since it saves some heap space and run time speed, we can wrap them both to standardize but comes with some exection costs.

Reply:
Ah, ok! Thanks for explaining!